.github/workflows/build.yaml

```diff
@@ -46,39 +37,34 @@ jobs:
         run: sudo su -c "PACKAGE_CLI=1 GITHUB_OUTPUT=$GITHUB_OUTPUT ./ci/scripts/build.sh"
         if: ${{ matrix.postgres == 15 }} # run only once
       - name: Upload archive package artifacts
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
-          name: lantern-extras-package
+          name: ${{ steps.build.outputs.archive_package_name }}
           path: ${{ steps.build.outputs.archive_package_path }}
-      - name: Upload cli package artifacts
-        uses: actions/upload-artifact@v3
+      - name: Upload Lantern CLI artifacts
         if: ${{ matrix.postgres == 15 }} # run only once
+        uses: actions/upload-artifact@v4
         with:
-          name: lantern-cli-package
+          name: ${{ steps.build_cli.outputs.cli_package_name }}
           path: ${{ steps.build_cli.outputs.cli_package_path }}
   ubuntu-package:
     runs-on: ubuntu-22.04
     needs: [ubuntu-build]
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/download-artifact@v3
+      - uses: actions/checkout@v4
+      - uses: actions/download-artifact@v4
         with:
-          name: lantern-extras-package
+          pattern: lantern-extras-*
+          merge-multiple: true
           path: /tmp/lantern-extras-package
-      - uses: actions/download-artifact@v3
+      - uses: geekyeggo/delete-artifact@v4
```

Comment:
just curious, what happened here? actions/download -> geekyeggo/delete ?

Reply:
I actually removed the delete action for CLI artifact, from here the packaging step, and let it upload from previous (build step).

This is what was happening before:
1. Build CLI (build step)
2. Upload CLI (build step)
3. Download CLI (package step)
4. Upload CLI with correct name (package step)

And this is what is happening now:
1. Build CLI (build step)
2. Upload CLI with correct name arch and platform appended (build step)

I didn't replace the delete action with download this is just github showing like that.

But I have also updated delete/download/upload/checkout actions to v4 to remove the depreciation warnings. I will do the same on lantern repo as well